src/meta/src/rpc/service/notification_service.rs

```diff
@@ -84,8 +84,13 @@ where
         let creating_tables = catalog_guard.database.list_creating_tables();
         let users = catalog_guard.user.list_users();
 
-        let cluster_guard = self.cluster_manager.get_cluster_core_guard().await;
-        let nodes = cluster_guard.list_worker_node(WorkerType::ComputeNode, Some(Running));
+        let context_id = self
+            .cluster_manager
+            .get_cluster_core_guard()
+            .await
+            .get_worker_by_host(host_address.clone())
+            .unwrap()
```

Comment:
get_worker_by_host may return None, if worker is removed before Line 86.

Reply:
> get_worker_by_host may return None, if worker is removed before Line 86. We can return Err in this case.

fixed